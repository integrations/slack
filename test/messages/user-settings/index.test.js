const { CombinedSettings } = require('../../../lib/messages/user-settings');


describe('CombinedSettings rendering', () => {
  test('works for both auto unfurl settings and muted prompts', async () => {
    const autoUnfurlRepos = [
      {
        id: 12345,
        full_name: 'integrations/snappydoo',
      },
      {
        id: 54321,
        full_name: 'integrations/html-to-mrkwdn',
      },
    ];
    const muteUnfurlPromptsIndefinitely = true;
    const attachments = new CombinedSettings({
      autoUnfurlRepos,
      muteUnfurlPromptsIndefinitely,
    }).getAttachments();
    expect({ attachments }).toMatchSnapshot();
    expect(attachments[0].actions[0].options[0].text).toBe('integrations/snappydoo');
    expect(attachments[0].actions[0].options[1].text).toBe('integrations/html-to-mrkwdn');
  });

  test('works for only auto unfurl settings', async () => {
    const autoUnfurlRepos = [
      {
        id: 12345,
        full_name: 'integrations/snappydoo',
      },
      {
        id: 54321,
        full_name: 'integrations/html-to-mrkwdn',
      },
    ];
    const attachments = new CombinedSettings({
      autoUnfurlRepos,
    }).getAttachments();
    expect({ attachments }).toMatchSnapshot();
  });

  test('works only for muted prompts', async () => {
    const muteUnfurlPromptsIndefinitely = true;
    const attachments = new CombinedSettings({
      muteUnfurlPromptsIndefinitely,
    }).getAttachments();
    expect({ attachments }).toMatchSnapshot();
  });

  test('works for temporarily muted prompts', async () => {
    Date.now = jest.fn(() => new Date(Date.UTC(2018, 4, 4)).valueOf());
    const muteUnfurlPromptsIndefinitely = false;
    const muteUnfurlPromptsUntil = 1525398771;
    const attachments = new CombinedSettings({
      muteUnfurlPromptsIndefinitely,
      muteUnfurlPromptsUntil,
    }).getAttachments();
    expect({ attachments }).toMatchSnapshot();
  });

  test('doesn\'t show if temporary mute is no longer active', async () => {
    Date.now = jest.fn(() => new Date(Date.UTC(2018, 4, 4)).valueOf());
    const muteUnfurlPromptsIndefinitely = false;
    const muteUnfurlPromptsUntil = 1522713600;
    const attachments = new CombinedSettings({
      muteUnfurlPromptsIndefinitely,
      muteUnfurlPromptsUntil,
    }).getAttachments();
    expect({ attachments }).toMatchSnapshot();
  });
});
