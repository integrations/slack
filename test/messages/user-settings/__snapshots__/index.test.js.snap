// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CombinedSettings rendering doesn't show if temporary mute is no longer active 1`] = `
Object {
  "attachments": Array [
    Object {
      "color": "#24292f",
      "text": "You do not have any settings configured.",
    },
  ],
}
`;

exports[`CombinedSettings rendering works for both auto unfurl settings and muted prompts 1`] = `
Object {
  "attachments": Array [
    Object {
      "actions": Array [
        Object {
          "name": "get-settings-for-repo",
          "options": Array [
            Object {
              "text": "integrations/snappydoo",
              "value": "12345|integrations/snappydoo",
            },
            Object {
              "text": "integrations/html-to-mrkwdn",
              "value": "54321|integrations/html-to-mrkwdn",
            },
          ],
          "text": "Select a repository",
          "type": "select",
        },
      ],
      "callback_id": "unfurl-settings-auto",
      "color": "#24292f",
      "text": "Links you post linking to 2 private repositories automatically get a rich preview. Select a repository to adjust its settings.",
      "title": "Automatic rich previews",
    },
    Object {
      "actions": Array [
        Object {
          "name": "unmute-prompts",
          "text": "Unmute",
          "type": "button",
        },
      ],
      "callback_id": "unfurl-settings",
      "color": "#24292f",
      "text": "Prompts to show a rich preview are muted indefinitely",
      "title": "Muted prompts to show rich preview",
    },
  ],
}
`;

exports[`CombinedSettings rendering works for only auto unfurl settings 1`] = `
Object {
  "attachments": Array [
    Object {
      "actions": Array [
        Object {
          "name": "get-settings-for-repo",
          "options": Array [
            Object {
              "text": "integrations/snappydoo",
              "value": "12345|integrations/snappydoo",
            },
            Object {
              "text": "integrations/html-to-mrkwdn",
              "value": "54321|integrations/html-to-mrkwdn",
            },
          ],
          "text": "Select a repository",
          "type": "select",
        },
      ],
      "callback_id": "unfurl-settings-auto",
      "color": "#24292f",
      "text": "Links you post linking to 2 private repositories automatically get a rich preview. Select a repository to adjust its settings.",
      "title": "Automatic rich previews",
    },
  ],
}
`;

exports[`CombinedSettings rendering works for temporarily muted prompts 1`] = `
Object {
  "attachments": Array [
    Object {
      "actions": Array [
        Object {
          "name": "unmute-prompts",
          "text": "Unmute",
          "type": "button",
        },
      ],
      "callback_id": "unfurl-settings",
      "color": "#24292f",
      "text": "Prompts to show a rich preview are muted for 2 hours",
      "title": "Muted prompts to show rich preview",
    },
  ],
}
`;

exports[`CombinedSettings rendering works only for muted prompts 1`] = `
Object {
  "attachments": Array [
    Object {
      "actions": Array [
        Object {
          "name": "unmute-prompts",
          "text": "Unmute",
          "type": "button",
        },
      ],
      "callback_id": "unfurl-settings",
      "color": "#24292f",
      "text": "Prompts to show a rich preview are muted indefinitely",
      "title": "Muted prompts to show rich preview",
    },
  ],
}
`;
